consensus/ethash/sealer.go

```diff
@@ -226,11 +231,15 @@ func (ethash *Ethash) remote(notify []string) {
 	// submitWork verifies the submitted pow solution, returning
 	// whether the solution was accepted or not (not can be both a bad pow as well as
 	// any other error, like no pending work or stale mining result).
-	submitWork := func(nonce types.BlockNonce, mixDigest common.Hash, hash common.Hash) bool {
+	submitWork := func(nonce types.BlockNonce, mixDigest common.Hash, sealhash common.Hash) bool {
 		// Make sure the work submitted is present
-		block := works[hash]
+		block := works[sealhash]
 		if block == nil {
-			log.Info("Work submitted but none pending", "hash", hash)
+			log.Info("Work submitted but none pending", "sealhash", sealhash, "current block number", currentBlock.NumberU64())
```

Comment:
Take care never to have a log context field with a space in it (`current block number`), it will just add weird quotation marks in the output. Perhaps you could use `curnumber` and `cursealhash` if you want to add a bit more context.

Perhaps we can also raise the level from Info to Warn? It's definitely not a happy path.